File: c/grains/src/grains.c
#include "grains.h"


uint64_t square(uint8_t index){
    
    return (index < 1 || index > 65) ? 0 : pow(2, index - 1);

}

uint64_t total(void){
    return 18446744073709551615ull;
}

File: go/grains/grains.go
package grains

import (
	"errors"
	"math"
)

func Square(input int) (uint64, error) {

	if input < 1 || input > 64 {
		return 0, errors.New("input must be between 1 and 64")
	}

	return uint64(math.Pow(2, float64(input-1))), nil

}

func Total() uint64 {
	return 18446744073709551615
}
